### ISSUE TITLE
[youtube] The following content is not available on this app.. Watch on the latest version of YouTube

### ISSUE THREAD

**Issue by saidjonUzz** [yt-dlp/yt-dlp#10108]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

There is such an error, what can be done?

### Provide verbose output that clearly demonstrates the problem

- [x] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [x] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.05.27 from yt-dlp/yt-dlp [12b248ce6] (zip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-107-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2019.11.28, mutagen-1.47.0, requests-2.31.0, sqlite3-3.31.1, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1820 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.05.27 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.05.27 from yt-dlp/yt-dlp)
[youtube] Extracting URL: https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U
[youtube] 3z0IT3bXm_E: Downloading webpage
[youtube] 3z0IT3bXm_E: Downloading ios player API JSON
ERROR: [youtube] 3z0IT3bXm_E: The following content is not available on this app.. Watch on the latest version of YouTube.
  File "/home/user/.local/bin/yt-dlp/yt_dlp/extractor/common.py", line 734, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/bin/yt-dlp/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
    self.raise_no_formats(reason, expected=True)
  File "/home/user/.local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
    raise ExtractorError(msg, expected=expected, video_id=video_id)
```

**Comment by bashonly** [comment]
that is not a verbose log

**Comment by saidjonUzz** [comment]
> that is not a verbose log

this full log

**Comment by bashonly** [comment]
> Run your yt-dlp command with `-vU` flag added (`yt-dlp -vU <your command line>`)
> Copy the **WHOLE** output (starting with `[debug] Command-line config`)

**Comment by saidjonUzz** [comment]
> yt-dlp -vU

`yt-dlp` -vU https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U
[debug] Command-line config: ['-vU', 'https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.05.27 from yt-dlp/yt-dlp [12b248ce6] (zip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-107-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2019.11.28, mutagen-1.47.0, requests-2.31.0, sqlite3-3.31.1, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1820 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.05.27 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.05.27 from yt-dlp/yt-dlp)
[youtube] Extracting URL: https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U
[youtube] 3z0IT3bXm_E: Downloading webpage
[youtube] 3z0IT3bXm_E: Downloading ios player API JSON
ERROR: [youtube] 3z0IT3bXm_E: The following content is not available on this app.. Watch on the latest version of YouTube.
  File "/home/user/.local/bin/yt-dlp/yt_dlp/extractor/common.py", line 734, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/bin/yt-dlp/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
    self.raise_no_formats(reason, expected=True)
  File "/home/user/.local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
    raise ExtractorError(msg, expected=expected, video_id=video_id)

**Comment by saidjonUzz** [comment]
Still not working, could it be a problem with the server?

**Comment by andremxmx** [comment]
same here

´´´
system@user:~/frey$ ./yt-dlp -U
Latest version: stable@2024.05.27 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.05.27 from yt-dlp/yt-dlp)
system@user:~/frey$ ./yt-dlp Q1kmXoeeKAE
[youtube] Extracting URL: Q1kmXoeeKAE
[youtube] Q1kmXoeeKAE: Downloading webpage
[youtube] Q1kmXoeeKAE: Downloading ios player API JSON
ERROR: [youtube] Q1kmXoeeKAE: The following content is not available on this app.. Watch on the latest version of YouTube.
system@user:~/frey$ ./yt-dlp -vU https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U
[debug] Command-line config: ['-vU', 'https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.05.27 from yt-dlp/yt-dlp [12b248ce6] (zip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.4.0-182-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg N-92517-gd0f594202a-Xtreamui-Mods-cesarkrespo (fdk)
[debug] Optional libraries: certifi-2019.11.28, requests-2.22.0, secretstorage-2.3.1, sqlite3-3.31.1, urllib3-1.25.8
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1820 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.05.27 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.05.27 from yt-dlp/yt-dlp)
[youtube] Extracting URL: https://youtu.be/3z0IT3bXm_E?si=sJlxgc-qRI8mxT3U
[youtube] 3z0IT3bXm_E: Downloading webpage
[youtube] 3z0IT3bXm_E: Downloading ios player API JSON
ERROR: [youtube] 3z0IT3bXm_E: The following content is not available on this app.. Watch on the latest version of YouTube.
  File "./yt-dlp/yt_dlp/extractor/common.py", line 734, in extract
    ie_result = self._real_extract(url)
  File "./yt-dlp/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
    self.raise_no_formats(reason, expected=True)
  File "./yt-dlp/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
    raise ExtractorError(msg, expected=expected, video_id=video_id)
´´´

**Comment by winf-hsos** [comment]
Same problem here:

`DownloadError: ERROR: [youtube] ... : The following content is not available on this app.. Watch on the latest version of YouTube.`

**Comment by burtek** [comment]
Works correctly for me on [latest nightly](https://github.com/yt-dlp/yt-dlp#update-channels) version

**Comment by triplesixman** [comment]
> Works correctly for me on [latest nightly](https://github.com/yt-dlp/yt-dlp#update-channels) version

I don't think you'll be affected by this problem, as there's been no change between the release and the nightly version. A priori I'm stuck on the last release and the nightly as well.

**Comment by Joel18241096** [comment]
same problem here

**Comment by burtek** [comment]
> > Works correctly for me on [latest nightly](https://github.com/yt-dlp/yt-dlp#update-channels) version
> 
> I don't think you're affected by this problem because there's been no change between the release and the nightly. I have the same error on the last release and the nightly as well.

So might be just YouTube doing some A/B testing, with B version not compatible with yt-dlp...

**Comment by winf-hsos** [comment]
It works for me on one computer, but not on another. I thought it might be the IP-addresses. Made many requests from the computer where it suddenly doesn't work anymore. Could it be I've been blocked?

**Comment by bashonly** [comment]
Yes, this is how youtube's IP banning is manifesting itself via yt-dlp output now

**Comment by TheSystemGuy1337** [comment]
> Yes, this is how youtube's IP banning is manifesting itself via yt-dlp output now

YouTube is harming **legitimate users and data preservation efforts** yet failing to address scammers. They must **REALLY** hate their user base if they are going this far. People are just going to work around these blocks; it's a waste of time. How they are still doing this baffles me. Probably because it harms their **bottom line.**

**Comment by kane50613** [comment]
Is this related? https://support.google.com/youtube/thread/269521462?hl=en

**Comment by TheSystemGuy1337** [comment]
> Is this related? https://support.google.com/youtube/thread/269521462?hl=en

I took one good look at it and it's a blatant "screw you" to their users. YouTube just decided to lock themselves into a battle that they cannot win to "protect the creators" even though they do next to nothing for the creators themselves [and allow scammers to run wild and free.](https://www.youtube.com/watch?v=Maeh4Qhhpwk) They must have been trying to aim at NewPipe and third-party YouTube clients and yt-dlp got caught in the crossfire or has been targeted directly. What's the point in trying to block third party clients when they can just patch their code to work around your blocks?

**Comment by Abedalhkeem-z** [comment]
Any solutation?

**Comment by triplesixman** [comment]
New error message :

```
ERROR: [youtube+NSigDeno] YVkUvmDQ3HY: You’ve hit the request limit.. Your limits will reset soon, usually within an hour.
  File "/opt/yt-dlp/yt_dlp/extractor/common.py", line 734, in extract
    ie_result = self._real_extract(url)
  File "/opt/yt-dlp/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
    self.raise_no_formats(reason, expected=True)
  File "/opt/yt-dlp/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
    raise ExtractorError(msg, expected=expected, video_id=video_id)

```

**Comment by TheSystemGuy1337** [comment]
> Same issue here. Would a proxy fix it ?

You can wait for your IP to change naturally or for YouTube to unblacklist your IP. Most people have dynamic IPs; it may change every week, or every month. A proxy or VPN will get around this.

**Comment by Technerder** [comment]
Getting something similar on my end, not sure if related

```
[youtube] Extracting URL: https://www.youtube.com/watch?v=c0_lGFdm_Ak
[youtube] Sleeping 1.0 seconds ...
[youtube] c0_lGFdm_Ak: Downloading webpage
WARNING: [youtube] Unable to download webpage: HTTP Error 429: Too Many Requests
[youtube] Sleeping 1.0 seconds ...
[youtube] c0_lGFdm_Ak: Downloading ios player API JSON
[youtube] Sleeping 1.0 seconds ...
[youtube] c0_lGFdm_Ak: Downloading iframe API JS
[youtube] Sleeping 1.0 seconds ...
[youtube] c0_lGFdm_Ak: Downloading web player API JSON
ERROR: [youtube] c0_lGFdm_Ak: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
  File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 734, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
    self.raise_no_formats(reason, expected=True)
  File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
    raise ExtractorError(msg, expected=expected, video_id=video_id)
```

**Comment by TheSystemGuy1337** [comment]
> Getting something similar on my end, not sure if related
> 
> ```
> [youtube] Extracting URL: https://www.youtube.com/watch?v=c0_lGFdm_Ak
> [youtube] Sleeping 1.0 seconds ...
> [youtube] c0_lGFdm_Ak: Downloading webpage
> WARNING: [youtube] Unable to download webpage: HTTP Error 429: Too Many Requests
> [youtube] Sleeping 1.0 seconds ...
> [youtube] c0_lGFdm_Ak: Downloading ios player API JSON
> [youtube] Sleeping 1.0 seconds ...
> [youtube] c0_lGFdm_Ak: Downloading iframe API JS
> [youtube] Sleeping 1.0 seconds ...
> [youtube] c0_lGFdm_Ak: Downloading web player API JSON
> ERROR: [youtube] c0_lGFdm_Ak: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
>   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 734, in extract
>     ie_result = self._real_extract(url)
>                 ^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
>     self.raise_no_formats(reason, expected=True)
>   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
>     raise ExtractorError(msg, expected=expected, video_id=video_id)
> ```

YouTube appears to be banning IPs fully now; "The following content is not available on this app. Watch on the latest version of YouTube." was just a coverup of what they were really trying to do. The only real fix is for the devs to start their own proxy and make yt-dlp use that proxy by default, so IP addresses are unique and yours won't get blacklisted. But starting a proxy is complicated and I don't really understand how a proxy works

**Comment by triplesixman** [comment]
> > Getting something similar on my end, not sure if related
> > ```
> > [youtube] Extracting URL: https://www.youtube.com/watch?v=c0_lGFdm_Ak
> > [youtube] Sleeping 1.0 seconds ...
> > [youtube] c0_lGFdm_Ak: Downloading webpage
> > WARNING: [youtube] Unable to download webpage: HTTP Error 429: Too Many Requests
> > [youtube] Sleeping 1.0 seconds ...
> > [youtube] c0_lGFdm_Ak: Downloading ios player API JSON
> > [youtube] Sleeping 1.0 seconds ...
> > [youtube] c0_lGFdm_Ak: Downloading iframe API JS
> > [youtube] Sleeping 1.0 seconds ...
> > [youtube] c0_lGFdm_Ak: Downloading web player API JSON
> > ERROR: [youtube] c0_lGFdm_Ak: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
> >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 734, in extract
> >     ie_result = self._real_extract(url)
> >                 ^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
> >     self.raise_no_formats(reason, expected=True)
> >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
> >     raise ExtractorError(msg, expected=expected, video_id=video_id)
> > ```
> 
> YouTube appears to be banning IPs fully now; "The following content is not available on this app. Watch on the latest version of YouTube." was just a coverup of what they were really trying to do. The only real fix is for the devs to start their own proxy and make yt-dlp use that proxy by default, so IP addresses are unique and yours won't get blacklisted. But starting a proxy is complicated and I don't really understand how a proxy works

They also ban residential IPs, so this solution isn't viable in my opinion.

**Comment by triplesixman** [comment]
`Sign in to confirm you’re not a bot. This helps protect our community. Learn more`

This error seems to appear randomly, I'd say 1 in 5 times.

**Comment by TheSystemGuy1337** [comment]
> > > Getting something similar on my end, not sure if related
> > > ```
> > > [youtube] Extracting URL: https://www.youtube.com/watch?v=c0_lGFdm_Ak
> > > [youtube] Sleeping 1.0 seconds ...
> > > [youtube] c0_lGFdm_Ak: Downloading webpage
> > > WARNING: [youtube] Unable to download webpage: HTTP Error 429: Too Many Requests
> > > [youtube] Sleeping 1.0 seconds ...
> > > [youtube] c0_lGFdm_Ak: Downloading ios player API JSON
> > > [youtube] Sleeping 1.0 seconds ...
> > > [youtube] c0_lGFdm_Ak: Downloading iframe API JS
> > > [youtube] Sleeping 1.0 seconds ...
> > > [youtube] c0_lGFdm_Ak: Downloading web player API JSON
> > > ERROR: [youtube] c0_lGFdm_Ak: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
> > >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 734, in extract
> > >     ie_result = self._real_extract(url)
> > >                 ^^^^^^^^^^^^^^^^^^^^^^^
> > >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
> > >     self.raise_no_formats(reason, expected=True)
> > >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
> > >     raise ExtractorError(msg, expected=expected, video_id=video_id)
> > > ```
> > 
> > 
> > YouTube appears to be banning IPs fully now; "The following content is not available on this app. Watch on the latest version of YouTube." was just a coverup of what they were really trying to do. The only real fix is for the devs to start their own proxy and make yt-dlp use that proxy by default, so IP addresses are unique and yours won't get blacklisted. But starting a proxy is complicated and I don't really understand how a proxy works
> 
> They also ban residential IPs, so this solution isn't viable in my opinion.
> 
> `Sign in to confirm you’re not a bot. This helps protect our community. Learn more`
> 
> It seems that this error is displayed randomly, I'd say 1 in 5 times, at least according to my tests.

If they suspect you are a machine, why don't they just toss HTTP 403 or HTTP 401?

**Comment by coletdjnz** [comment]
Related: https://github.com/iv-org/invidious/issues/4734

**Comment by kane50613** [comment]
> > > > Getting something similar on my end, not sure if related
> > > > ```
> > > > [youtube] Extracting URL: https://www.youtube.com/watch?v=c0_lGFdm_Ak
> > > > [youtube] Sleeping 1.0 seconds ...
> > > > [youtube] c0_lGFdm_Ak: Downloading webpage
> > > > WARNING: [youtube] Unable to download webpage: HTTP Error 429: Too Many Requests
> > > > [youtube] Sleeping 1.0 seconds ...
> > > > [youtube] c0_lGFdm_Ak: Downloading ios player API JSON
> > > > [youtube] Sleeping 1.0 seconds ...
> > > > [youtube] c0_lGFdm_Ak: Downloading iframe API JS
> > > > [youtube] Sleeping 1.0 seconds ...
> > > > [youtube] c0_lGFdm_Ak: Downloading web player API JSON
> > > > ERROR: [youtube] c0_lGFdm_Ak: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
> > > >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 734, in extract
> > > >     ie_result = self._real_extract(url)
> > > >                 ^^^^^^^^^^^^^^^^^^^^^^^
> > > >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 4248, in _real_extract
> > > >     self.raise_no_formats(reason, expected=True)
> > > >   File "/home/user/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1257, in raise_no_formats
> > > >     raise ExtractorError(msg, expected=expected, video_id=video_id)
> > > > ```
> > > 
> > > 
> > > YouTube appears to be banning IPs fully now; "The following content is not available on this app. Watch on the latest version of YouTube." was just a coverup of what they were really trying to do. The only real fix is for the devs to start their own proxy and make yt-dlp use that proxy by default, so IP addresses are unique and yours won't get blacklisted. But starting a proxy is complicated and I don't really understand how a proxy works
> > 
> > 
> > They also ban residential IPs, so this solution isn't viable in my opinion.
> > `Sign in to confirm you’re not a bot. This helps protect our community. Learn more`
> > It seems that this error is displayed randomly, I'd say 1 in 5 times, at least according to my tests.
> 
> If they suspect you are a machine, why don't they just toss HTTP 403 or HTTP 401?

They will throw a 403 error when you try to access the `/videoplayback` endpoint returned by the player.

**Comment by TheSystemGuy1337** [comment]
> They will throw a 403 error when you try to access the `/videoplayback` endpoint returned by the innertube `/player` endpoint.

If that's the case, then yt-dlp should just throw `FATAL: Your IP has been blocked by YouTube.` followed by HTTP 403 rather than `Sign in to confirm you’re not a bot.`

**Comment by kane50613** [comment]
I'm curious whether it judges your behavior or simply looks at the number of IP requests. 

For example, the browser might also send requests to the `/api/stats/watchtime` endpoint.

**Comment by Wack0** [comment]
Probably kinda offtopic but I'm actually interested in where that "learn more" link goes...

**Comment by TheSystemGuy1337** [comment]
> I'm curious whether it judges your behavior or simply looks at the number of IP requests.
> 
> For example, the browser might also send requests to the `/api/stats/watchtime` endpoint.

Judging by behavior (users report they get this block on one computer, but other systems are unaffected), it's likely an IP blacklist. Download too many times and it will trip an automated system resulting in this block. We could slow downloads down to reduce the chances of tripping the system. And to Wack0, it likely leads you to some nonsense blog post filled with tons of corporate speak on how they are trying to "protect the creators" when they are doing **NOTHING** for them unless you are a major public figure, a celebrity or someone major on YouTube.

**Comment by velw** [comment]
So does it go away if you're logged in on the browser and pass the browser cookies as a parameter? (Would test but am not in a position to do so right now — just want to throw the idea out there in case it helps someone.)

**Comment by 00-kat** [comment]
>  pass the browser cookies as a parameter?

Might wanna see [this](https://github.com/imputnet/cobalt/issues/551). Using browser cookies is probably a terrible idea.

**Comment by velw** [comment]
> Might wanna see [this](https://github.com/imputnet/cobalt/issues/551). Using browser cookies is probably a terrible idea.

Thank you - sorry, I hadn't seen that (or I wouldn't have suggested it). Yes it does sound like a terrible idea 🤦 

Interested in how they're identifying that yt-dlp requests are not the same as regular requests from their app or the browser 🤔 (presuming the same thing doesn't happen if you watch too many videos in your browser)

**Comment by TheSystemGuy1337** [comment]
> Interested in how they're identifying that yt-dlp requests are not the same as regular requests from their app or the browser 🤔 (presuming the same thing doesn't happen if you watch too many videos in your browser)

You may have potentially found the key to working around this blockade that YouTube set up. yt-dlp may need to imitate and forge proper requests. But that may still not work; and the best solution is to avoid the block in the first place which sadly may mean slowing the connection or setting a counter for how many downloads you can do per day. It will cripple yt-dlp but at least work around the banning.

**Comment by 00-kat** [comment]
> Thank you - sorry, I hadn't seen that (or I wouldn't have suggested it). Yes it does sound like a terrible idea 🤦

That was in a different repo, so you probably wouldn't have found it anyways. No need to apologise

> Interested in how they're identifying that yt-dlp requests are not the same as regular requests from their app or the browser

My guess is either user agent, or what Kane suggested above
> the browser might also send requests to the /api/stats/watchtime endpoint

Although ad blockers exist, so who knows

**Comment by 00-kat** [comment]
> slowing the connection or setting a counter for how many downloads you can do per day.

Keep in mind Invidious, Piped, and other alternate frontends are experiencing this too. One of the people on the bug report in the Invidious one said it happened on their private instances as well, with barely anyone using it. A maximum download count is probably not the main issue here, or maybe it's a combination of issues.

**Comment by wukko** [comment]
> it happened on their private instances as well, with barely anyone using it

that's because it's (seemingly) ip-based a/b testing. cloud provider's ips are probably more likely to be affected than residential ips.

**Comment by 00-kat** [comment]
Oh true, I forgot that small instances tend to be run on VPSs. Maybe it could just be a maximum download count then.

**Comment by TheSystemGuy1337** [comment]
> Oh true, I forgot that small instances tend to be run on VPSs. Maybe it could just be a maximum download count then.

If you don't download frequently or download with self-restraint then you should be unaffected. This is only affecting those who download frequently. Those who download in bulk or download playlists have the greatest risk of facing this block. YouTube is harming both personal, non-commercial use of YouTube videos and data preservation efforts.

**Comment by velw** [comment]
> client does not matter, even youtube's own web client will give you a sign in error if your ip (in this case of a vpn) isn't deemed good enough: [TeamNewPipe/NewPipe#11139 (comment)](https://github.com/TeamNewPipe/NewPipe/issues/11139#issuecomment-2154637732)

An edit to that comment says the same video then plays again without demanding a login if the IP address changes (VPN is disabled). So it seems like what's critical here is not causing the IP address to get blocked in the first place.

I think there's a reasonable case for personal noncommercial downloads of YouTube videos and content from other streaming sites — not everyone is blessed with the kind of bleeding edge bandwidth so many sites demand to watch content at a good quality — and for data preservation as well.

If someone is downloading massive quantities of content quickly, though, I can see a reasonable case for whoever runs that infrastructure to want the requests to be throttled so it doesn't deny service to others. I know that's only the fig leaf YouTube would use here to justify what their motives really are, but the fig leaf in itself isn't entirely unreasonable. 

Is it worth having some kind of throttling as the default setting (at least until this is figured out) and letting the user remove it with a command switch? Or have that as the default for whole playlist downloads, even if not for individual video downloads?

**Comment by NipLars** [comment]
If it really was about stopping bots Google could just throw their "I'm not a robot" recaptha.
- you can configure that to do really abusive tests as well as rather modest

So I agree with those thinking this is purely to stop downloaders. This benefit both Google and those producing content worth throwing ads at, if they manage to do that.

And if logged into account at Google they can do more statistics if they get people to sign in more.

But incredible devs at yt-dlp manage to work around things before, so probably will again.

Going with Brave browser give the same "sign in....bot"-suggestion so it's not just yt-dlp, it's general problem and wanting people to sign in more, as well, as I see it.
- and just on some files, not all
- I went back and forth a couple of times, and only certain files give "bot"-message

**Comment by TheSystemGuy1337** [comment]
> If it really was about stopping bots Google could just throw their "I'm not a robot" recaptcha.
> 
> * you can configure that to do really abusive tests as well as rather modest
> 
> So I agree with those thinking this is mostly to stop downloaders. This benefit both Google and those producing content worth throwing ads at, if they manage to do that.
> 
> And if logged into account at Google they can do more statistics if they get people to sign in more.
> 
> But incredible devs at yt-dlp manage to work around things before, so probably will again.
> 
> Going with Brave browser give the same "sign in....bot"-suggestion so it's not just yt-dlp, it's general problem and wanting people to sign in more, as well, as I see it.
> 
> * and just on some files, not all
> * I went back and forth a couple of times, and only certain files give "bot"-message
> 
> And I do vpn all the time, and swapping servers and it might go through in the end. And all cookies and history is cleared in Brave every time I start it. So new ip all the time and still get it mostly on some files.

And the only reason why they are stopping downloaders is because it harms their bottom line. They went on a crusade against adblock and failed. This issue with them blocking downloaders is already sparking controversy as it broke music bots on Discord and it won't be long before this (hopefully) reaches the mainstream.

**Comment by NipLars** [comment]
Hopefully Google find that side effects are worse and getting less views on content and thereby less ads played so they retract the idea.

That browsers don't get access either will create resentment from content producers as well.

The cure become worse than the disease.

**Comment by NipLars** [comment]
**Is this a clue, maybe?**

Using an embedded youtube video it works like here
https://riks.se/riksronden-ska-eu-forskare-och-experter-styra-sverige/

but if I search that on youtube and access it need login.
- either just clicking to watch it directly on youtube or search separately on youtube
- as does yt-dlp also does

**To emulate being embedded might be a way to download it?**

Just a thought....

**Comment by 00-kat** [comment]
> **To emulate being embedded might be a way to download it?**

Note that the site you linked uses youtube.com and not youtube-nocookie.com (or whatever the site was), so it's using your account's cookies if you're signed in. In case you didn't already, you might want to try a private window.

**Comment by triplesixman** [comment]
> > **To emulate being embedded might be a way to download it?**
> 
> Note that the site you linked uses youtube.com and not youtube-nocookie.com, so it's using your account's cookies if you're signed in. In case you didn't already, you might want to try a private window.

A priori what he said seems correct.

My IP is currently blocked too, so if I'm not logged in I can't play videos.

I've tried going to the site without private browsing and with (without being logged into an account in both cases) and the video plays.

**Comment by NipLars** [comment]
> > **To emulate being embedded might be a way to download it?**
> 
> Note that the site you linked uses youtube.com and not youtube-nocookie.com, so it's using your account's cookies if you're signed in. In case you didn't already, you might want to try a private window.

I have not been signed into youtube for last 5 years, so I am not now.

I tried the /embed/ url thingy from view source, but that was not enough.

So something how it's embedded make a difference it seems.

**And again - it's even with browser it refuses, not only yt-dlp**

**Comment by coletdjnz** [comment]
> **Is this a clue, maybe?**
> 
> Using an embedded youtube video it works like here with browser
> https://riks.se/riksronden-ska-eu-forskare-och-experter-styra-sverige/
> 
> but if I search that on youtube and access it need login.
> - either just clicking to watch it directly on youtube or search separately on youtube
> - as does yt-dlp also does needing sign in
> 
> **To emulate being embedded might be a way to download it?**
> 
> Just a thought....

You could try test this by setting innertube_host to the YouTube no cookie domain

And set the player_client to web_embedded

**Comment by MoralCode** [comment]
i would be curious whether this change affects things like DuckDuckGo's youtube video embeds when you search for video results - if they somehow get an exception to this maybe that could lead to a workaround

**Comment by NipLars** [comment]
I have tested arbitrary search Duck, if you mean some content show youtube vids in a rectangle.
- there is no difference clicking those, but those become the usuall youtube pages
But I might be unaware of what search you use if some plugin from Duck?

**Comment by NipLars** [comment]
I tested and installed DuckDuckGo-browser now and still no go without sign in picking Duck player.

Maybe this Google strike is about how they introduce ad free youtube and such?
- not so much downloaders

**Comment by NipLars** [comment]
I tested further and seems to be about ip-addresses used and identified as malicious or similar
- the regular vpn servers I use are all blocked
- but picking one I never used before like Andorra went through with Duck player at least
**So vpn servers used by bots might be targeted and you as private user can be affected too.**

It happends that protective scripts show a recaptcha to go through at various places using these scripts.
- sometimes message as title of this thread
- sometime just that you are blocked
**Vpn services are used by bots as well and create issues for all of us.**

**Comment by TheSystemGuy1337** [comment]
[The developers of Cobalt have found a workaround, but it involves OAuth. Unless people are willing to volunteer and create dummy accounts and a dozen OAuth keys then encrypt and obfuscate them and include it with yt-dlp to defeat this bullshit then we'll be stuck here until another workaround is discovered or Google backs down, which is unlikely..](https://github.com/imputnet/cobalt/issues/551#issuecomment-2156024956)

**Comment by seproDev** [comment]
Experimental OAuth2 support for YouTube already exists for yt-dlp with https://github.com/coletdjnz/yt-dlp-youtube-oauth2

**Comment by sivannavis** [comment]
Having the same problem because of running it on ssh remote, the above answer using oauth2 solved it!

**Comment by triplesixman** [comment]
> Experimental OAuth2 support for YouTube already exists for yt-dlp with https://github.com/coletdjnz/yt-dlp-youtube-oauth2

Does oauth require you to log in/enter the code regularly? I'm not familiar with this method and would like to know whether I should automate tasks or do it once manually.

**Comment by coletdjnz** [comment]
> > Experimental OAuth2 support for YouTube already exists for yt-dlp with https://github.com/coletdjnz/yt-dlp-youtube-oauth2
> 
> Does oauth require you to log in/enter the code regularly? I'm not familiar with this method and would like to know whether I should automate tasks or do it once manually.

It's still experimental and not fully battle-tested what happens in the long term for yt-dlp. YouTube provides a long-lived refresh token which has been enough to keep yt-dlp logged in so far in my experience.


It's the same method used to log into YouTube on TV

**Comment by TheSystemGuy1337** [comment]
> It's still experimental and not fully battle-tested what happens in the long term for yt-dlp. However YouTube provides a long-lived refresh token which has been enough to keep yt-dlp logged in so far in my experience.
> 
> It's the same method used to log into YouTube on TV

It might not also be risk-free, judging by Google's behavior they may terminate your account or ban you in some way. Either that or they will quietly brick OAuth by screwing with YouTube on TV. A risk-free workaround is still needed, but at least we have one workaround.

**Comment by velw** [comment]
> yt-dlp still worked for me yesterday but I doubt it's going to work for me in a week before I face this issue. I am in TX and don't download often. Maybe I will never face this issue? Who knows.

Might be useful if people who are reporting being affected by this can note roughly how many downloads they've done in the day or week before hitting the block, so it's possible to tell whether occasional downloaders are affected (i.e. probably everyone, sooner or later) or a smaller group.

**Comment by TheSystemGuy1337** [comment]
I'll keep track. I have downloaded about 2 or 3 videos today as far as my memory goes

**Comment by triplesixman** [comment]
oauth2 doesn't seem to be a solution for downloading large numbers of youtube videos.

`You’ve hit the request limit.. Your limits will reset soon, usually within an hour.`

**Comment by velw** [comment]
> You’ve hit the request limit.. Your limits will reset soon, usually within an hour.

At least it gave some more information. What happened after an hour?

**Comment by spirillen** [comment]
OT: to @NipLars 

I can read you are using the Brave browser, May I recommend you to switch to Firefox-esr for security and better privacy? By using Brave browser all your traffic is interrupted by Brave.

See also: https://github.com/badmojr/1Hosts/issues/1783 (among many)

**Comment by TheSystemGuy1337** [comment]
How is this related to the current discussion of YouTube banning IPs because they are greedy and don’t care about their users?

**Comment by spirillen** [comment]
> How is this related to the current discussion of YouTube banning IPs because they are greedy and don’t care about their users?

OT = Off Topic...

**Comment by bashonly** [comment]
This is **not** directed at anyone in particular, but this needs to be said before this issue thread gets completely unreadable & unmanageable:

### An issue tracker is NOT a message board.

Even though Github's issues UI looks essentially like a phpBB forum, this is not a phpBB forum.

Off-topic comments do not belong here. Rants against Youtube/Google/Brave/etc do not belong here (even if they are deserved). If a comment has been marked as off-topic or otherwise hidden, do not engage with it.

"Me too"/"+1" comments are not useful after the issue has already been established to be widespread. Speculation based on evidence/history is OK, but wild speculation is misinformation.

### Take the chit-chat to [discord](https://discord.gg/H5MNcFW63r).

Issue threads need to be able to function as **resources of information** for:
1. developers trying to fix the issue
2. other impacted users looking for solutions/workarounds

Maintainers can hide irrelevant comments, but maintainers can't undo Github's automatic hiding of potentially useful comments once the thread reaches a certain length. And the longer the thread gets, the more comments are automatically hidden by Github.

### Please only comment if you have new/useful information pertaining this issue. Thanks in advance.

**Comment by Solsticely** [comment]
have we considered also downloading an ad and waiting the length of the ad before buffering the main video? (hell maybe even feeding some dummy analytics to guise as a normal browser?)

implementing this stuff sounds like it'd be quite the feat but it might help with not triggering youtube's ytdlp detection in the long run, and we can just put it behind a command line argument for people who don't want to wait and are okay with risking being blocked

**Comment by agieocean** [comment]
> have we considered also downloading an ad and waiting the length of the ad before buffering the main video? (hell maybe even feeding some dummy analytics to guise as a normal browser?)
> 
> implementing this stuff sounds like it'd be quite the feat but it might help with not triggering youtube's ytdlp detection in the long run, and we can just put it behind a command line argument for people who don't want to wait and are okay with risking being blocked

I've seen people say setting the user agent to edge helps avoid the a/b testing but I haven't tested that with yt-dlp

**Comment by Lcchy** [comment]
> have we considered also downloading an ad and waiting the length of the ad before buffering the main video? (hell maybe even feeding some dummy analytics to guise as a normal browser?)

When viewing videos through youtube.com with an adblocker like uBlock, I don't get ads and I also don't get the "please login" block.

That would indicate that the detection actually doesn't take into account the ad-playing delay ?

**Comment by stop-treachery** [comment]
The fact reusing browser cookies seem to cause account blocks to happen much faster should be a hint. What is the browser doing with those cookies that yt-dlp is not? In particular, are the same cookies getting used more than once? Does youtube's JS rewrite the cookies each time to something new (i.e. a one-time pad, but obtainable somehow via the same methods used to calculate every other needed parameter)?

**Comment by stop-treachery** [comment]
In other words, in addition to whatever yt-dlp does differently than youtube.com in accessing videos when logged out, what's different when logged in about the cookie handling, since that specifically seems to be detected?

**Comment by NipLars** [comment]
A couple of things browser do

Moving from one page to another the http header Referer hold the url you come from click some link.
- doing open in another tab does not do this

Using youtube search engine since a year or so you get a pp= parameter which encodes what the search words were. So youtube can map what you searched before clicking on an url.
- I remove that before using yt-dlp not to inform Google where url came from

I mostly use a file with one or many urls with yt-dlp, this can also be noticed by Google thinking you are a bot.

There may be loads of criteria Google use to finally flag an ip.

Bots often use vpn and flag the ip no for us using it for just ordinary browsing and some downloads.
- 8 servers I used the most are now flagged because too much bot traffic I assume
- this is now a major issue for vpn providers I am sure

**Don't know why Google is flagging bots, unless they load the system seriously
- but Google have bots all the time updating the search engine
- and other search engines would have bots also updating youtube stuff there**

I get the feeling this is about something else than the "bot" thing. This is just an excuse to track more statistics about users or something being signed in.

**Comment by stop-treachery** [comment]
It seems to be easier/faster to get accounts blocked, than IPs, though. Which means there is something about the cookies and logged-in state specifically that is worth investigating.

I have doubts on referer; I haven't noticed any issues using YouTube with a live browser and I often am in a private/incognito tab.

Obviously VPNs are going to be an issue (as they already are for privacy apps for other social networks) given the much higher volume. How does Google tell the difference between a previously unknown VPN IP and a CGNAT IP, given how much CGNAT junk is going on these days?

**Comment by coletdjnz** [comment]
This is likely related to BotGuard which YouTube is A/B testing/rolling out currently.

**Comment by Lcchy** [comment]
@coletdjnz is there a public channel where we could follow these discussions?

I totally understand if not, I am just interested out of curiosity.

**Comment by velw** [comment]
Am still keen to hear what happened after the hour mentioned in the limit warning with oauth2, and also how many videos people are downloading before they run into this problem. @TheSystemGuy1337 mentioned they'd downloaded 2 or 3 but it wasn't clear whether they encountered the block immediately after that. If the block can be lifted after an hour, and you have to download a really large number of videos before it's applied, that makes addressing this a different priority to if it's affecting people after only a few downloads and the block isn't lifted after the hour.

**Comment by NipLars** [comment]
Is it something to do with DuckDuckGo own browser also allowing embedded viewing without the ads?
- the core of Google is threatened and trying to remedy anything doing that

Until 3 years ago I ran and tried to keep an XP Windows machine running by having User-Agent manager in browser avoiding ban from many places due to how server intercept that.
- since it's easy to circumvent Google probably don't rely too much on that is my guess

I experimented a bit with "Referer:" in yt-dlp before I realized browsers were not allowed either.
- trying to make yt-dlp look more like a browser
- but no go still

The Cookie popup dialog I mostly do Reject All, so they keep track other ways than cookies
- but try Accept All too and no difference with browser

**Some places like logging into Microsoft.com and my account does not allow doing that over VPN at all, so think there is some way to determine if VPN ip.**

**Comment by 00-kat** [comment]
> Is it something to do with DuckDuckGo own browser also allowing embedded viewing without the ads?

I should mention (not necessarily replying to your comment) that on Android, DuckDuckGo browser just uses webview. That means if your webview implementation is set to the default, it's using Chrome. The only other webview implementations I know are the Android one (also Chrome) and the Bromite one. Since acceptible webview implementations are hardcoded, you need root to use ones like the Bromite one. I.e. it could be something related to Chrome rather than DuckDuckGo's browser.


> * since it's easy to circumvent Google probably don't rely too much on that is my guess

If the user agent is something like curl it is probably blocked. There's no way that's the only defense, but it can't hurt to have it there as an additional one


> The Cookie popup dialog I mostly do Reject All, so they keep track other ways than cookies

Probably browser fingerprinting. Anyways that cookie popup, if you read it, is only about using cookies for advertising purposes and the like. There's always some cookies used for other purposes that can't be disabled.


> **Some places like logging into Microsoft.com and my account does not allow doing that over VPN at all, so think there is some way to determine if VPN ip.**

VPN IPs tend to look like datacenters. I highly doubt a human would try to login to microsoft.com from a datacenter.
> forza.net

However, a web scraper (search engine, ...) might.

**Comment by TheSystemGuy1337** [comment]
> Am still keen to hear what happened after the hour mentioned in the limit warning with oauth2, and also how many videos people are downloading before they run into this problem. @TheSystemGuy1337 mentioned they'd downloaded 2 or 3 but it wasn't clear whether they encountered the block immediately after that. If the block can be lifted after an hour, and you have to download a really large number of videos before it's applied, that makes addressing this a different priority to if it's affecting people after only a few downloads and the block isn't lifted after the hour.


I have not faced this block yet and I’ve downloaded around 10 to 15 videos over the past week. For whatever reason I tried replying to this via email and it didn't go though, so if it dupes blame Gmail.

**Comment by tr13dm03** [comment]
Besides the number of videos downloaded within an hour, concurrency of downloads (and of their fragments) and rate-limiting or lack thereof may also have an influence.

I strongly rate-limit my downloads (<10Mbps) since a few years when Youtube started aggressively throttling them if I didn't and download exclusively serially, one at a time, and I've yet to be affected by this current issue (besides the various Invidious instances I used no longer working).

**Comment by nikita2498** [comment]
It doesn't matter if you use OAuth instead of cookies, it seems to be putting a complete ban on user who downloads "too much". I have recently tried to download a playlist through plugins and it gave me this:
`ERROR: [youtube+oauth2] <videoID>: Video unavailable. This content isn't available.` It blocked my throwaway account from downloading and watching videos. But before that i was getting `Skipping player response` on [https://www.youtube.com/watch?v=aQvGIIdgFDM](this video) so that was apparently A/B block testing before completely shutting any downloads on JS level...

**Comment by NipLars** [comment]
Another OAuth experience I had the other day, as I was to file a bug report at Mozilla.org.
- it gave option to login with Github
- as I did that is said no account based on that email address and created one

**I really thought all credentials were to be protected, but revealed which email Github was based on.**

Just something to think about, if anybody think their email is safe. It's probably only pwd.

**Comment by TheSystemGuy1337** [comment]
UPDATE: I was able to get away with downloading ten videos in a row without encountering the block. Either I am extremely lucky, YouTube isn't targeting people in the Midwestern United States or are not targeting Starlink customers. 

However, I'm laying down the baseline that one video is fine, five is pushing it and ten+ is risky. The more you download, the more likely you may face this block.

**Comment by agieocean** [comment]
> UPDATE: I was able to get away with downloading ten videos in a row without encountering the block. Either I am extremely lucky, YouTube isn't targeting people in the Midwestern United States or are not targeting Starlink customers.
> 
> However, I'm laying down the baseline that one video is fine, five is pushing it and ten+ is risky. The more you download, the more likely you may face this block.

Think this might have something to do with residential ips, datacenter ips seem to get this issue far more often at least in my experience running a downloader site

**Comment by Faeb35** [comment]
Go the "The following content is not available on this app.. Watch on the latest version of YouTube" error. 
Adding --cookies and --user-agent solved it for me.

**Comment by TheSystemGuy1337** [comment]
DO NOT PASS COOKIES. YOU WILL GET COOKIEBANNED.

**Comment by kiwib1rd1337** [comment]
I got this error just now with my private YT video archiving project (from New Zealand ISP, no login cookies are used).
Passing -4 to force IPv4 mode seems to have fixed it for now (successfully fetched around 50 videos in a row on IPv4, IPv6 threw the "not available on this app" error).

My dynamic IPv6 prefix updates every day or so, and I haven't been downloading in almost half a week, and no other hosts on my network have this issue (Network has /64 prefix).

My ISP has dual-stack, where the IPv4 is one of those CG-NAT jobs (horay for IPv4 address exhaustion). Blocking IPv4 in my situation may cause collateral damage on YT's part.

I think they might be fingerprinting hosts based on IPv6 address, probably the EUI-64 part the contains the MAC address (those cheeky little monkeys). The EUI-64 is consistent, even with a dynamic prefix. This can prove really effective at tracking devices, even when they move between different networks.
Source: https://www.theregister.com/2022/03/22/legacy_ipv6_addressing_standard_enables/

A possible mitigation would be have users look at enabling "IPv6 privacy extensions" (i.e. rfc4941 and rfc7824) if they don't have it enabled already. This should mitigate this tracking somewhat. Spoofing the MAC address may be another one.
For static IPv6 prefixes, that may be more complicated (they may ban the prefix also).

YT better be careful if they are doing this in the EU (MAC address may fall under GDPR protection).
Anyone else have this situation, try forcing IPv4 mode to confirm this EUI-64 ban situation?

**Comment by ryuqe787** [comment]
Hello everyone! I've encountered this issue too, when I was downloading videos from multiple user accounts for archival purposes. I have a fixed IP address, so changing it isn't an option for me. I was looking for a quick, free solution. I don't want to advertise, but the free version of ProtonVPN completely solved the problem. I use it on an Ubuntu server with an ovpn file, since there's no GUI. It's also a benefit that the VPN connection reduces the bandwidth. I'd like to ask if there's a way to create a service that disconnects the existing VPN connection and takes the next VPN config file to establish a new connection upon receiving a specific error signal. Does yt-dlp have a return code that would indicate this type of error separately? Thanks in advance!

**Comment by TheSystemGuy1337** [comment]
> Hello everyone! I've encountered this issue too, when I was downloading videos from multiple user accounts for archival purposes. I have a fixed IP address, so changing it isn't an option for me. I was looking for a quick, free solution. I don't want to advertise, but the free version of ProtonVPN completely solved the problem. I use it on an Ubuntu server with an ovpn file, since there's no GUI. It's also a benefit that the VPN connection reduces the bandwidth. I'd like to ask if there's a way to create a service that disconnects the existing VPN connection and takes the next VPN config file to establish a new connection upon receiving a specific error signal. Does yt-dlp have a return code that would indicate this type of error separately? Thanks in advance!

There is no such thing as a "free" VPN. Nothing in life is free, and they are likely selling your data to recuperate losses. Thats if they don't throttle you to like 56K or have a 500 MiB data limit. Even then, this is far beyond the scope of this issue. [Please read this. You're making it harder for the yt-dlp devs to actually fix this issue.](https://github.com/yt-dlp/yt-dlp/issues/10108#issuecomment-2159720541)

**Comment by Tokyo-All-Ready** [comment]
I dont see any other workaround then just proxying/vpn it. Using yt-dlp WILL blacklist your ip, i am not sure for how long.  Also do not use --cookies-from-browser, it will load so slowly if you use a web client then speeds up after about 3-4 hours.

**Comment by bashonly** [comment]
Closing since there's nothing we can do
